### ISSUE TITLE
Allow setting default item attachment

### ISSUE THREAD

**Issue by aurimasv** [zotero/zotero#355]
This has been requested before, but it was mostly ignored because there was a limited use case for this and potential for user confusion. (e.g. thread https://forums.zotero.org/discussion/29638/how-to-set-default-pdf-to-be-displayed-via-view-pdf/)

Well, I'm actually getting burned by this myself today. I have an item with a PDF for the paper and a PDF for the supplementary. Originally everything was fine, because the paper PDF was added before the supplementary PDF, which means the right PDF would open. Unfortunately (mostly due to my clumsiness), I ended up deleting the paper PDF. Now if I re-add the PDF, the supplementary data opens by default. I'd have to remove the supplementary PDF and add it from either local disk or the webpage again.

Here's what I propose (as mentioned in the linked thread).

When a child attachment is selected, in the right-hand pane, under Filename add a line that says "Default attachment" or "Open by default" with a checkbox next to it. Checking this box would make that attachment open up when double-clicking the item. Unchecking it, would revert to Zotero's default behavior.

The default state of the checkbox would be unchecked (possibly confusing, because via Zotero's default mechanism, this could still be the default attachment). Alternatively (to avoid the mentioned confusion), this could be a radio button that could not be unchecked (only transfered to another attachment).

Only one of the attachments for an item could have the checkbox or radio button checked.

If this sounds reasonable, then I would be happy to submit a patch. My only question would be whether to store this default attachment assignment in a new table or an additional column in "itemAttachments" table (doesn't look like an attachment can be part of two different items.. which, I suppose, makes sense).

**Comment by zirydy** [comment]
I know it's been a while since this proposal has been reported, but I would like to see this implemented as well. I sometimes keep more than one PDF for a book, and it would be nice if it was possible to choose which to open by default.

**Comment by 0Hughman0** [comment]
I'm 100% for adding this feature, for the exact reason stated. Adding supplementary notes to a database entry causes the main article to no longer open by default - surely this is something that scientific researchers (the main users of this software... I imagine), would run into constantly?

**Comment by dstillman** [comment]
@0Hughman0: No, because that's not the behavior. It should continue to open the first PDF added to the item. If you're seeing something different, report it in the Zotero Forums.

**Comment by jix** [comment]
I have another use case for this, which AFAICT was not mentioned so far. When adding a paper via the browser extension, zotero often automatically downloads the corresponding arxiv preprint if the paper itself is not available as open access from the publisher. If I then later add the final version of the paper, there seems to be no way to make zotero open the final version without deleting the the arxiv preprint (which I'd like to keep, so I can later check if there are any changes between the versions).

**Comment by dejavumono** [comment]
@jix I had the same use case. Typically, I would add a arxiv preprint manually and read it. After final version comes out, I would append it. For now, there is no easy way to open the final version as default.

**Comment by mikeblazanin** [comment]
Just wanted to add I also would find this feature useful, since for many papers I have both the main text and the supplement, which may not be in the correct order.

I also wonder if maybe making it possible to reorder the attachments by clicking and dragging them within the parent item, and having the pdf viewer by-default open the top item, would be more intuitive than having a checkbox somewhere denoting one of the attachments as default? Haven't thought through all the possible pros/cons of that approach vs radio button or checkbox approach, but perhaps something to consider

**Comment by PierreENOlivier** [comment]
It is now 2023, and I would definitely see this feature added. :)
As @jix mentionned, it very often happen that Zotero would download an Arxiv preprint automatically. Adding the final print later on would still open the first attachment by default.
